transform/mattermost-analytics/models/marts/product/licenses/_product__licenses__models.yml

```diff
@@ -91,3 +91,65 @@ models:
         description: |
           Whether the expiration date is consistent in case there's multiple sources that report the license. True if 
           more than one expiration dates are reported, false else.
+
+  - name: dim_self_hosted_customers
+    description: |
+      List of known self-hosted customers with active or inactive licenses along with their server details. 
+      Note that the same customer/server might appear more than once in case the details are different for the given license.
+      We are de-duplicating servers based on the latest telemetry received, there will be only 1 row per server_id
+
+    columns:
+      - name: self_hosted_customer_id
+        description: A unique identifier for the self_hosted customer generated from server_id and license_id.
+        tests:
+          - not_null
+          - unique
+      - name: server_id
+        description: The server id of the license.    
+        tests:
+          - not_null
+          - unique
```

Comment:
what should I remove the `unique`, `not_null` constraint on here?

Reply:
> If self_hosted_customer_id is preferred over server_id, then this id must be present in the fact table in order to allow for join. Will that guarantee that each self_hosted_customer_id in fact table also exists in the dimension table?

This is not present in the fact model, I added this, because I thought we require a surrogate key for every dimension model, but looks like the surrogate key purpose is for a primary key. In that case, `server_id` is good here

Reply:
A primary key is a good idea so that joins can happen on that key. Sometimes a surrogate key might be the primary key.

Reply:
Removed the surrogate keys.

Reply:
I understand the concept of surrogate key, but in this particular use case, the data is coming from 2 different sources, 1 is license telemetry and the other is NPS (which might or might not contain the `license_id` and does not have the `installation_id`) because of which we cannot use the surrogate key to join here. I added the surrogate key to the NPS fact model, since every marts model need to have a primary key, but it won't be used for any joining.